src/fsharp/FSharp.Core/tasks.fs

```diff
@@ -0,0 +1,372 @@
+    open Microsoft.FSharp.Core.LanguagePrimitives.IntrinsicOperators
+
+    module LowPriority = 
+        // Low priority extensions
+        type TaskBuilderBase with
+
+            static member inline BindDynamic< ^TaskLike, ^TResult1, 'TResult2, ^Awaiter , 'TOverall
+                                                when  ^TaskLike: (member GetAwaiter:  unit ->  ^Awaiter)
+                                                and ^Awaiter :> ICriticalNotifyCompletion
+                                                and ^Awaiter: (member get_IsCompleted:  unit -> bool)
+                                                and ^Awaiter: (member GetResult:  unit ->  ^TResult1)>
+                        (sm: byref<_>, task: ^TaskLike, continuation: (^TResult1 -> TaskCode<'TOverall, 'TResult2>)) : bool =
+
+                    let mutable awaiter = (^TaskLike: (member GetAwaiter : unit -> ^Awaiter)(task)) 
+
+                    let cont = 
+                        (TaskResumptionFunc<'TOverall>( fun sm -> 
+                            let result = (^Awaiter : (member GetResult : unit -> ^TResult1)(awaiter))
+                            (continuation result).Invoke(&sm)))
+
+                    // shortcut to continue immediately
+                    if (^Awaiter : (member get_IsCompleted : unit -> bool)(awaiter)) then 
+                        cont.Invoke(&sm)
+                    else
+                        sm.ResumptionDynamicInfo.ResumptionData <- (awaiter :> ICriticalNotifyCompletion)
+                        sm.ResumptionDynamicInfo.ResumptionFunc <- cont
+                        false
+
+            member inline _.Bind< ^TaskLike, ^TResult1, 'TResult2, ^Awaiter , 'TOverall
+                                                when  ^TaskLike: (member GetAwaiter:  unit ->  ^Awaiter)
+                                                and ^Awaiter :> ICriticalNotifyCompletion
+                                                and ^Awaiter: (member get_IsCompleted:  unit -> bool)
+                                                and ^Awaiter: (member GetResult:  unit ->  ^TResult1)>
+                        (task: ^TaskLike, continuation: (^TResult1 -> TaskCode<'TOverall, 'TResult2>)) : TaskCode<'TOverall, 'TResult2> =
+
+                TaskCode<'TOverall, _>(fun sm -> 
+                    if __useResumableCode then 
+                        //-- RESUMABLE CODE START
+                        // Get an awaiter from the awaitable
+                        let mutable awaiter = (^TaskLike: (member GetAwaiter : unit -> ^Awaiter)(task)) 
+
+                        let mutable __stack_fin = true
+                        if not (^Awaiter : (member get_IsCompleted : unit -> bool)(awaiter)) then
+                            // This will yield with __stack_yield_fin = false
+                            // This will resume with __stack_yield_fin = true
+                            let __stack_yield_fin = ResumableCode.Yield().Invoke(&sm)
```

Comment:
In the spec of resumable code, you need to use

    let v = <invoke resumable code>
    v2 <- v

rather than

    v2 <- <invoke resumable code>

That is resumable code must follow these very limited patterns where the control structure is not hidden.